packages/backend/server/src/plugins/copilot/session.ts

```diff
@@ -495,6 +500,35 @@ export class ChatSessionService {
     });
   }
 
+  async cleanup(
+    options: Omit<ChatSessionOptions, 'promptName'> & { sessionIds: string[] }
+  ) {
+    return await this.db.$transaction(async tx => {
+      const sessions = await tx.aiSession
+        .findMany({
+          where: {
+            id: { in: options.sessionIds },
+            userId: options.userId,
+            workspaceId: options.workspaceId,
+            docId: options.docId,
+            deletedAt: null,
+          },
+          select: { id: true },
+        })
+        .then(sessions => sessions.map(({ id }) => id));
+      // cleanup all messages
+      await tx.aiSessionMessage.deleteMany({
+        where: { sessionId: { in: sessions } },
+      });
+      // only mark action session as deleted
+      // chat session always can be reuse
+      await tx.aiSession.updateMany({
+        where: { id: { in: sessions }, prompt: { action: { not: null } } },
```

Comment:
the join query would be super slow. how do you think bringing a little redundant data like action to ai sessions table

Reply:
now to use local prompt cache to check, this info basically not be modified, so it is safe enough to use cache first